src/metabase/api/cache.clj

```diff
@@ -148,4 +150,48 @@
                              nil)))
   nil)
 
+(api/defendpoint POST "/invalidate"
+  "Invalidate cache entries"
+  [include database dashboard question]
+  {include   [:maybe [:= :overrides]]
+   database  [:maybe (ms/QueryVectorOf ms/IntGreaterThanOrEqualToZero)]
+   dashboard [:maybe (ms/QueryVectorOf ms/IntGreaterThanOrEqualToZero)]
+   question  [:maybe (ms/QueryVectorOf ms/IntGreaterThanOrEqualToZero)]}
+
+  (when-not (premium-features/enable-cache-granular-controls?)
+    (throw (premium-features/ee-feature-error (tru "Granular Caching"))))
+
+  (if-not (= include :overrides)
+    (let [conditions (for [[k vs] [[:database database]
+                                   [:dashboard dashboard]
+                                   [:question question]]
+                           v      vs]
+                       [:and [:= :model (name k)] [:= :model_id v]])
+          cnt        (when (seq conditions)
+                       ;; using JVM date rather than DB time since it's what are used in cache tasks
+                       (t2/update! CacheConfig {:id [:in {:from   (t2/table-name CacheConfig)
+                                                          :select [:id]
+                                                          :where  (into [:or] conditions)}]}
+                                   {:invalidated_at (t/offset-date-time)}))]
```

Comment:
Is there a better way to pass raw honeysql condition to a `t2/update!`? I did not find anything suitable.